code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -173,16 +174,19 @@
 
 		else
 			if(!status)
-				L.visible_message("<span class='warning'>[L] has been prodded with [src] by [user]. Luckily it was off.</span>")
+				L.visible_message("<span class='warning'>[L] has been prodded with \the [src] by [user]. Luckily it was off.</span>")
 				return 1
 			else
-				L.visible_message("<span class='danger'>[L] has been prodded with [src] by [user]!</span>")
+				L.visible_message("<span class='danger'>[L] has been prodded with \the [src] by [user]!</span>")
 
 	//stun effects
 	L.stun_effect_act(stun, agony, target_zone, src)
 
 	playsound(loc, 'sound/weapons/Egloves.ogg', 50, 1, -1)
-	msg_admin_attack("[key_name_admin(user)] stunned [key_name_admin(L)] with the [src] (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",ckey=key_name(user),ckey_target=key_name(L))
+	if(!no_attack_log && user.a_intent != I_HURT)
```

Comment:
What is the purpose of this check?

Reply:
Because from code above when we are in Harm intent , it calls parent proc which already logs in VV just in different format. This check avoid making duplicates.

Reply:
Then why not move this into the "else" clause of that check? It'd make more sense then.